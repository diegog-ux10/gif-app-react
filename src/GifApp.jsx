import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball'])

    const onAddCategory = (newCategory) => {
        const arrCheck = categories.map( (cat) => {return cat.toLocaleLowerCase} );
        const valCheck = newCategory.toLowerCase
        if(arrCheck.includes(valCheck)) return;
        setCategories([newCategory, ...categories]);
    }

    return (
        <>
            <h1>Gif App</h1>
            <AddCategory
                onNewCategory={(value) => onAddCategory(value)} 
            />
            <button onClick={ onAddCategory }>Agregar</button>
            <ol>
                {categories.map(category => (
                    <GifGrid key={category} category={category}/>
                ))}
            </ol>
        </>
    )
}