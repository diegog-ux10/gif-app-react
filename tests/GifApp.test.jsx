import { render, renderHook, screen, fireEvent } from "@testing-library/react";
import { GifApp } from "../src/GifApp";
import { useState } from "react";

describe('Test on <GifApp />', () => {
    test('should match snapshot', () => {
        const {container} = render(<GifApp />);
        expect(container).toMatchSnapshot();
    })

    test('should change the categories', () => {
        const {result} = renderHook(()=> useState(['One Punch', 'Dragon Ball']));
        const [categories] = result.current;
        const inputValue = 'Messi';

        render(<GifApp />);

        const input = screen.getByRole('textbox');
        fireEvent.input( input, {target: {value: inputValue}} );
        
        expect(input.value).toBe(inputValue)
    })
});
