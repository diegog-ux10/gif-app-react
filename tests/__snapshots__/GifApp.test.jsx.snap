// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifApp /> should match snapshot 1`] = `
<div>
  <h1>
    Gif App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gif"
      type="text"
      value=""
    />
  </form>
  <button>
    Agregar
  </button>
  <ol>
    <h3>
      One Punch
    </h3>
    <h2>
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
    <h3>
      Dragon Ball
    </h3>
    <h2>
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
